app/services/proposal_service.rb

```diff
@@ -0,0 +1,55 @@
+class ProposalService
+  # include Rails.application.routes.url_helpers
+  # AO_TOKEN_LIFETIME = 3
+
+  # def initialize(token_service = nil, current_time = nil)
+  #   @token_service = token_service || TokenService.new
+  #   @current_time = current_time || DateTime.now.utc
+  # end
+
+  def propose(form)
+    raise ArgumentError, 'form is invalid' unless form.valid?
+
+    ActiveRecord::Base.transaction do
+      pq = build_pq(form)
+
+      form.action_officer_id.uniq.each do |id|
+        action_officer = ActionOfficer.find id
```

Comment:
`find` will cause an exception if the action officer is not found, so the check on the next line is redundant.
If you use `find_by_id` it will return nil if not found which might be what you want here.

Reply:
If you add a test for this it might help you decide what is best here.